src/models/BaseModel.py

```diff
@@ -1,118 +1,205 @@
+        print(len(set(processed_job_titles)))
+        print(len(set(processed_job_modifiers)))
+        print(len(set(self.title_feature)))
+        # Job category: Binary encoded
+        processed_category_feature = remove_sub_string('Jobs', self.category_feature)
+        binary_encoded_categories = get_binary_encoded_feature(processed_category_feature)
+
+        concatanated_features = np.concatenate((
+                    onehot_encoded_desc_words,
+                    onehot_encoded_contract_type,
+                    onehot_encoded_contract_time,
+                    binary_encoded_company,
+                    binary_encoded_source,
+                    binary_encoded_town,
+                    binary_encoded_region,
+                    binary_encoded_job_titles,
+                    binary_encoded_job_modifiers,
+                    binary_encoded_categories,
+                    [pandas_vector_to_list(self.salary_feature)]
+        ))
+        concatanated_features = np.transpose(concatanated_features)
+
+        feature_list = [('desc_words', len(onehot_encoded_desc_words)),
+                        ('contract_type', len(onehot_encoded_contract_type)),
+                        ('contract_time', len(onehot_encoded_contract_time)),
+                        ('company', len(binary_encoded_company)),
+                        ('source', len(binary_encoded_source)),
+                        ('town', len(binary_encoded_town)),
+                        ('region', len(binary_encoded_region)),
+                        ('job_titles', len(binary_encoded_job_titles)),
+                        ('job_modifiers', len(binary_encoded_job_modifiers)),
+                        ('categories', len(binary_encoded_categories)),
+                        ]
+        print('Pre-processing ends \n')
+        return (concatanated_features, feature_list)
 
     @staticmethod
-    def load_location_data(train_normalized_location_file_name):
-        BaseModel.normalized_location_data = pd.read_csv(train_normalized_location_file_name)
+    def export_data(list_to_write, file_name):
+        print('Exporting data to ../data/'+file_name)
+        print('Will read from there next time to avoid pre-processing')
+        f = open('../data/'+file_name+'.csv', 'a')
+        for main_index, sub_list in enumerate(list_to_write):
+            for index, item in enumerate(sub_list):
+                f.write(str(item))
+                if index < len(sub_list)-1:
+                    f.write(',')
+            if main_index < len(list_to_write)-1:
+                f.write('\n')
+        f.close()
+        print('Exporting complete \n')
 
     @staticmethod
-    def load_csv_data(train_data_csv_file_name):
-        BaseModel.data = pd.read_csv(train_data_csv_file_name)
+    def export_prediction(prediction_to_write, file_name):
+        print('Exporting prediction to ../predictions/'+file_name)
+        f = open('../predictions/'+file_name+'.csv', 'a')
```

Comment:
might want to create the directory before writing to it, just for convenience for other people running this